eth/eventservices/rewardservice.go

```diff
@@ -18,6 +19,7 @@ var (
 
 type RewardService struct {
 	client       eth.LivepeerEthClient
+	pendingTx    *types.Transaction
```

Comment:
Would it make sense to make `pendingTx` a local variable, and pass it into the `tryReward` call, then return the result? This is mostly a matter of personal preference, but I try to avoid storing too much mutable state in structs (one step removed from simply making the state global).

Reply:
Hm `tryReward()` would need to tell the consumer to stop tracking a pending tx (if the pending tx was already mined) which would happen when it gets an `ErrReplacingMinedTx` error and to start tracking a pending tx (if the tx did not confirm within a defined time window) when it gets a `context.DeadlineExceeded` error. Tracking this via mutable state in a struct seems simpler and maybe ok since the `RewardService` struct is pretty small and single purpose at the moment. Let me know if you think avoiding mutable state in structs is worthwhile here though

Reply:
Yeah, the way to do it would be to propagate any pending tx via the return value, eg `tryReward(*types.Transaction) (*types.Transaction, error)` then feed the result back into `tryReward` for the next iteration.

Anyway it's just a personal preference stemming from the functional school of thought, so feel free to keep the current setup if you prefer. 🚢